src/providers/info.rs

```diff
@@ -188,15 +192,19 @@ impl<C: Cache> InfoProvider for DefaultInfoProvider<'_, C> {
 }
 
 impl<'a> DefaultInfoProvider<'a, ServiceCache<'a>> {
-    pub fn new(context: &'a Context) -> Self {
+    pub fn new(chain_id: &'a str, context: &'a Context) -> Self {
         DefaultInfoProvider {
+            chain_id,
             client: context.client(),
             cache: context.cache(),
-            chain_cache: Default::default(),
             safe_cache: Default::default(),
             token_cache: Default::default(),
         }
     }
+
+    pub fn new_no_net(context: &'a Context) -> Self {
+        Self::new("No network", context)
+    }
```

Comment:
There is a need for this method when we have information that is not network independant provided by this dependency. Namely, the fiat exchange rates. Are you ok with this @rmeissner ? It would just be a bit time consuming to adjust the tests again.

Reply:
Wondering if we can make `chain_id` as an optional field in this case. 

Or have a `ChainInfoProvider` and an `InfoProvider` (where `chain_id` is not available). I'd prefer this over `FiatInfoProvider` because having `Fiat` in the name might hint that only fiat is chain independent (where there might be more chain Independent fields that can be represented.

So we'd have `InfoProvider` (with chain independent data) and `ChainInfoProvider → InfoProvider` which includes chain data. Wdyt?

Reply:
So that is very similar to what I proposed. In the current state `DefaultInfoProvider` is the `ChainInfoProvider` from your proposal and `InfoProvider` is the `trait` that gets implemented by it. In my proposal `FiatInfoProvider` would be `InfoProvider` of your proposal.

I am just thinking about where we should make the cut, in terms of responsibilities. Do we distinguish chain and non chain data? For me that creates kind of only 2 groups of possible data sources, but if we wanted to have more, like for instance `TokenInfoProvider`, `ContractsInfoProvider`, etc. the `ChainInfoProvider` would get more bloated (kind of what we already have with `DefaultInfoProvider`).

 I would prefer the approach where we split this up by "topic" rather than being stuff that is chain related or not. I am just wondering if there is more fiat related stuff that we could use this for eventually. There is also an argument to be made in terms of "single responsibility" of a class. Admittedly, `DefaultInfoProvider` violates this currently.

Reply:
I've pushed this approach, seems clean and nice, but if for some reason we prefer the one commented on, we can revert the commit.